File: 1493-longest-subarray-of-1s-after-deleting-one-element/1493-longest-subarray-of-1s-after-deleting-one-element.py
class Solution(object):
    def longestSubarray(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        res = 0
        left = 0
        countZero = 0

        for right in range(len(nums)):
            if nums[right] == 0:
                countZero += 1
            
            while countZero == 2:
                if nums[left] == 0:
                    countZero -= 1
                left += 1

            res = max(res, right - left)

        return res

File: 1493-longest-subarray-of-1s-after-deleting-one-element/1493-longest-subarray-of-1s-after-deleting-one-element.go
func longestSubarray(nums []int) int {
    return longestOnes(nums, 1) - 1
}

func longestOnes(nums []int, k int) int {
    ans := 0
    l, r := 0, 0
    cnt := 0
    for r < len(nums) {
        if nums[r] == 0 {
            cnt += 1
            for cnt > k {
                if nums[l] == 0 {
                    cnt -= 1
                }

                l += 1
            }
        }

        r += 1

        ans = max(ans, r - l)
    }

    return ans
}